### DISCUSSION TITLE
Does older Windows support the latest PowerShell

### DISCUSSION THREAD

**Discussion by super-tongyao** [PowerShell/PowerShell#17648] [General]
### Summary of the new feature / enhancement

For example, does Windows7 support the installation of the latest version of PowerShell, can use SSH?

**Comment by jhoneill** [comment]
The word "support" here will cause all manner of problems

As used by Microsoft "Support" means there is an ultimate backstop of "If a bug is found, engineering resources will be used to produce a fix".  If PowerShell 7.2 exposes a bug in Windows 7, the Windows team will not (usually) go back and produce a fix for Windows to correct it - exceptions _are_ made but usually only if there is a security aspect. 

Will PowerShell _work_ as expected on out of support versions of Windows ? Quite possibly, but the PowerShell team won't test on any versions which are out of support, so if there is something they do which fails on older Windows they won't change their code either.

Generally if you are using something which forces you to use out-of-support Windows (e.g. some piece of hardware or software which can't work with anything else) best practice is to consider that system frozen, and to take steps to protect it from exposure to threats which which new systems are defended from.

**Comment by kasini3000** [comment]
1 you can install powershell 7.2 on win7 or win2008r2
2 you can install ssh on  win7 or win2008r2 .
https://github.com/PowerShell/Win32-OpenSSH/releases/download/v8.9.1.0p1-Beta/OpenSSH-Win64-v8.9.1.0.msi

3 from win to linux use **psremoting over ssh**, it is difference with **ssh**. 
psremoting can Using multithreading, .net objects can be passed between local and remote.

-----------------

[kasini3000](https://gitee.com/chuanjiao10/kasini3000) site mirror:(https://github.com/kasini3000/kasini3000)   
win,linux devops automation batch script framework.(It is similar to Puppet，Ansible，pipeline) 
Open source, free, cross-platform 
English manual: https://gitee.com/chuanjiao10/kasini3000/blob/master/docs/kasini3000_manual.xlsx

[kasini3000_agent_linux](https://gitee.com/chuanjiao10/kasini3000_agent_linux) Shell script,one click install powershell on linux,modify sshd_config for PsRemote.

1 Remote command line：(from win,linux master,to win,linux node)
$object_return_from_node_to_master = ./k_run_ip -ip 1.1.1.1 -scriptblock {command_on_node} -allparameter $a,$b
$object_return_from_node_to_master = ./k_run_ip -ip 1.1.1.1 -file 'd:\script_file_on_master.ps1' -allparameter $a,$b
$string_return_from_node_to_master = ./k_run_ip -ip 1.1.1.1 -scriptblock { bash /tmp/xxx.sh} -allparameter $a,$b

**Comment by kilasuit** [comment]
@kasini3000 - I have to ask you as one of the PowerShell Working group members to please stop promoting your product on issues and discussions as **this is not** the place to promote them.

**Comment by kilasuit** [comment]
This is not really an issue so I am converting this to a discussion.
